### DISCUSSION TITLE
How can I center an item?

### DISCUSSION THREAD

**Discussion by duxv** [AllenDang/giu#196] [Q&A] [answered]
I'm having trouble with centering a multiline user input, can somebody tell me how to?

**Comment by ItsVoltz** [comment]
Made a post in February about this have yet to get a response, still haven't figured it out coming back to my project after months

**Reply by AllenDang** [reply]
Refer #199

**Reply by gucio321** [reply]
@duxv try recently added `AlignSetter`

**Comment by gucio321** [comment, marked as answer]
@duxv @ItsVoltz I hope, you've already fixed it but if not:
if you mean centering items, you can use giu's AlignSetter BUT as it is an EXPERIMENTAL feature not supported by DearImgui, u may have trouble using it. AlignSetter should be great for centering labels and another widgets user'll not interract with, but for buttons, inpubfields e.t.c. I'd suggest using manual alignment.
The following example should demonstrate you usage of both: AlignSetter and the manual alignment
```golang
package main

import "github.com/AllenDang/giu"

var (
	username string
	password string
)

func loop() {
	giu.SingleWindow().Layout(
		giu.Align(giu.AlignCenter).To(
			giu.Label("Please enter username and password to XYZ"),
		),
		giu.Custom(func() {
			const inputFieldW = 200

			spacingX, _ := giu.GetItemSpacing()
			availableW, _ := giu.GetAvailableRegion()

			giu.Layout{
				giu.Row(
					giu.Dummy((availableW-inputFieldW)/2-spacingX, 0),
					giu.InputText(&username).Hint("Username").Size(inputFieldW),
				),
				giu.Row(
					giu.Dummy((availableW-inputFieldW)/2-spacingX, 0),
					giu.InputText(&password).Hint("Password").Flags(giu.InputTextFlagsPassword).Size(inputFieldW),
				),
			}.Build()
		}),
	)
}

func main() {
	wnd := giu.NewMasterWindow("Manual Alignment [example]", 640, 480, 0)
	wnd.Run(loop)
}

```
